src/callbacks_step/save_solution_2d_manifold_in_3d_cartesian.jl

```diff
@@ -0,0 +1,168 @@
+@muladd begin
+#! format: noindent
+
+# Convert to another set of variables using a solution_variables function
+function convert_variables(u, solution_variables, mesh::P4estMesh{2},
+                           equations::AbstractEquations{3}, dg, cache)
+    (; contravariant_vectors) = cache.elements
+    # Extract the number of solution variables to be output 
+    # (may be different than the number of conservative variables) 
+    n_vars = length(Trixi.varnames(solution_variables, equations))
+
+    # Allocate storage for output
+    data = Array{eltype(u)}(undef, n_vars, nnodes(dg), nnodes(dg), nelements(dg, cache))
+
+    # Loop over all nodes and convert variables, passing in auxiliary variables
+    Trixi.@threaded for element in eachelement(dg, cache)
+        for j in eachnode(dg), i in eachnode(dg)
+            normal_vector_node = Trixi.get_contravariant_vector(3,
+                                                                contravariant_vectors,
+                                                                i, j, element)
+
+            if applicable(solution_variables, u, normal_vector_node, equations, dg,
```

Comment:
I could not find a better way to "dispatch" on the arguments of `solution_variables`. The use of `try` (before the call to `convert_variables`) and `catch` (for the standard behavior) affects performance. Luckily, this alternative with `applicable()` does not affect performance.

Reply:
I hadn't heard of `applicable` before! If there is no effect on performance, it is fine with me.

Reply:
Here is the benchmark from the `main` branch for `cons2prim`:
```julia
julia> @benchmark Trixi.save_solution_file($u, 0.0, 0.1, 0, $mesh, $equations, $solver, $semi.cache, $save_solution.affect!)
BenchmarkTools.Trial: 1113 samples with 1 evaluation.
 Range (min … max):  3.252 ms … 27.921 ms  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     4.323 ms              ┊ GC (median):    0.00%
 Time  (mean ± σ):   4.484 ms ±  1.658 ms  ┊ GC (mean ± σ):  0.35% ± 0.91%

                ▃▄▂▄█▇▄▄█▆▅▂▃▃▄▁▅▃▂▂▄                         
  ▂▂▂▃▂▁▃▃▂▃▃▃█████████████████████████▇▇▅▃▄▃▃▁▄▂▂▁▁▁▂▂▁▁▁▁▂ ▅
  3.25 ms        Histogram: frequency by time         5.8 ms <

 Memory estimate: 297.60 KiB, allocs estimate: 499.
```
And here for the new branch:
```julia
julia> @benchmark Trixi.save_solution_file($u, 0.0, 0.1, 0, $mesh, $equations, $solver, $semi.cache, $save_solution.affect!)
BenchmarkTools.Trial: 1939 samples with 1 evaluation.
 Range (min … max):  1.341 ms … 34.306 ms  ┊ GC (min … max): 0.00% … 34.00%
 Time  (median):     2.467 ms              ┊ GC (median):    0.00%
 Time  (mean ± σ):   2.568 ms ±  1.772 ms  ┊ GC (mean ± σ):  0.81% ±  1.28%

              ▅██▅▅▅▄▂▁▂ ▃▄▅▅▄▇▆▄▂▅▄▅█▇▅▁▃▂▁▁                 
  ▃▃▂▂▁▂▃▁▃▂▂▄████████████████████████████████▅▆▄▄▃▅▃▃▃▃▃▂▂▂ ▅
  1.34 ms        Histogram: frequency by time        3.62 ms <

 Memory estimate: 203.74 KiB, allocs estimate: 496.
```